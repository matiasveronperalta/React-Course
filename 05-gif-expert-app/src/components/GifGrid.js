import React from 'react'

export const GifGrid = ({category}) => {

    const getGifs = async() =>{
        const url = 'https://api.giphy.com/v1/gifs/search?q=rick+and+morty&limit=10&api_key=UkXtWw1t02SZK27KypfrzFJ3wf5sWLnz';
        const response = await fetch(url);
        const {data} = await response.json();

        const gifs = data.map( img => {
            return{
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url,
            }
        });

        console.log(gifs);
    }

    getGifs();

    return (
        <>
            <h3>{category}</h3>
        </>
    )
}
