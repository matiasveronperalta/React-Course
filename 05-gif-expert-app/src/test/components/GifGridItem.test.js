import React from 'react';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect'
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';


describe('Test <GifGridItem/> ', () => {
    
    const title = 'ElTitulo';
    const url = 'https://localhost/noexiste.jpg'
    
    let wrapper = shallow(<GifGridItem title={title} url={url} />);

    beforeEach(() => {
        wrapper = shallow(<GifGridItem title={title} url={url} />);
    });

    test('should be show and match component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should be contain a pagraf with title', () => {
        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);
    })

    test('should be have image/url equals to props', () => {
        
        const image = wrapper.find('img');

        expect(image.prop('src')).toBe(url);
        expect(image.prop('alt')).toBe(title);

    })


    test('should be have animate__fadeIn style on div', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes('animate__fadeIn')).toBe(true);
    })
    
    
    

    
})
