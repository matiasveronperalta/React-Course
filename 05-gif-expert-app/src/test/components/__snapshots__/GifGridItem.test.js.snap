// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGridItem/>  should be show and match component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="ElTitulo"
    src="https://localhost/noexiste.jpg"
  />
  <p>
    ElTitulo
  </p>
</div>
`;
